plugins/auto-refetch/src/index.ts

```diff
@@ -0,0 +1,78 @@
+import type { PiniaColadaPlugin } from '@pinia/colada'
+
+export interface PiniaColadaAutoRefetchOptions {
+  /**
+   * Whether to enable auto refresh by default.
+   * @default true
+   */
+  enabled?: boolean
+}
+
+/**
+ * Plugin that automatically refreshes queries when they become stale
+ */
+export function PiniaColadaAutoRefetch(
+  options: PiniaColadaAutoRefetchOptions = {},
+): PiniaColadaPlugin {
+  const { enabled = true } = options
+
+  return ({ queryCache }) => {
+    // Keep track of active entries and their timeouts
+    const refetchTimeouts = new Map<string, NodeJS.Timeout>()
+
+    queryCache.$onAction(({ name, args, after }) => {
+      // We want refetch to happen only on the client
+      if (!import.meta.client) return
+
+      // Handle fetch to set up auto-refetch
+      if (name === 'refresh') {
```

Comment:
`refresh()` is indeed called more often than `fetch()`. Maybe we need a combination of `fetch` and `ensure` to make it work at creation too

Reply:
Makes sense. I refactored to ensure + fetch now. But I had to remove the `entry.active` check because `ensure` hook returns `useQueryOptions` instead of `UseQueryEntry`, but I am guessing that should be fine.